frontend/public/components/secrets/create-secret/PullSecretCredentialsForm.tsx

```diff
@@ -0,0 +1,111 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { AUTHS_KEY, PullSecretCredentialEntry, PullSecretData } from '.';
+import { usePullCredentialsFormSecretEntries } from './usePullCredentialsFormSecretEntries';
+
+const newImageSecretEntry = (): PullSecretCredential => ({
+  address: '',
+  username: '',
+  password: '',
+  email: '',
+  auth: '',
+  uid: _.uniqueId(),
+});
+
+export const PullSecretCredentialsForm: React.FC<PullSecretCredentialsFormProps> = ({
+  onChange,
+  pullSecretData,
+}) => {
+  const { t } = useTranslation();
+  const [secretEntriesArray, setSecretEntriesArray] = usePullCredentialsFormSecretEntries(
```

Comment:
we can simplify this since we already have the context of the component to imply what kind of form entries these are:
```suggestion
  const [entries, setEntries] = usePullCredentialsFormSecretEntries(
```